dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -690,67 +691,177 @@ def complete_load(self, load_id: str) -> None:
+
+        for table in table_lineage:
+            fq_table_name: str = self._job_client.make_qualified_table_name(table["name"])
+            try:
+                fq_parent_table_name: str = self._job_client.make_qualified_table_name(
+                    table["parent"]
+                )
+            except KeyError:
+                fq_parent_table_name = None  # The table is a root table.
+
+            try:
+                child_table = db_client.open_table(fq_table_name)
+                child_table.checkout_latest()
+                if fq_parent_table_name:
+                    parent_table = db_client.open_table(fq_parent_table_name)
+                    parent_table.checkout_latest()
+            except FileNotFoundError as e:
+                raise DestinationTransientException(
+                    "Couldn't open lancedb database. Orphan removal WILL BE RETRIED"
+                ) from e
+
+            try:
+                if fq_parent_table_name:
+                    # Chunks and embeddings in child table.
+                    parent_ids = set(
+                        pc.unique(
+                            parent_table.to_lance().to_table(columns=["_dlt_id"])["_dlt_id"]
+                        ).to_pylist()
+                    )
+                    child_ids = set(
+                        pc.unique(
+                            child_table.to_lance().to_table(columns=["_dlt_parent_id"])[
+                                "_dlt_parent_id"
+                            ]
+                        ).to_pylist()
+                    )
+
+                    if orphaned_ids := child_ids - parent_ids:
+                        if len(orphaned_ids) > 1:
+                            child_table.delete(f"_dlt_parent_id IN {tuple(orphaned_ids)}")
+                        elif len(orphaned_ids) == 1:
+                            child_table.delete(f"_dlt_parent_id = '{orphaned_ids.pop()}'")
+
+                else:
+                    # Chunks and embeddings in the root table.
+                    document_id_field = get_columns_names_with_prop(table, DOCUMENT_ID_HINT)
+                    if document_id_field and get_columns_names_with_prop(table, "primary_key"):
+                        raise SystemConfigurationException(
+                            "You CANNOT specify a primary key AND a document ID hint for the same"
+                            " resource when using merge disposition."
+                        )
+
+                    # If document ID is defined, we use this as the sole grouping key to identify stale chunks,
+                    # else fallback to the compound `id_field_name`.
+                    grouping_key = document_id_field or id_field_name
+                    grouping_key = (
+                        grouping_key if isinstance(grouping_key, list) else [grouping_key]
+                    )
+                    child_table_arrow: pa.Table = child_table.to_lance().to_table(
```

Comment:
please never use constant names for columns. you have normalized names in "schema" instance
schema.dlt_id_name (AFAIK) look for it

Reply:
Changed as requested, now reads from normalizer